src/indexing.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    CheckpointOutput
+
+This is a index entry describing the output file from a checkpoint.
+You can retrieve a list of these from a folder full of such outputs, using
+[`index_checkpoint_files`](@ref).
+
+For accessing details of the CheckpointOutput the following helpers are provided:
+[`path`](@ref), [`name`](@ref), [`groups`](@ref), [`tags`](@ref).
+Further: `getproperty` is overloaded so that you can access the value of the tag `:foo` via
+`x.foo`.
+"""
+struct CheckpointOutput{P<:AbstractPath}
```

Comment:
Why "Output" and not "Index"?

Reply:
Because the the index is the collection of all of these.
This described just one output.
I guess can call it a `CheckpointIndexEntry`
or even just a `IndexEntry`